src/psyclone/tests/psyir/nodes/dynamic_omp_task_directive_test.py

```diff
@@ -0,0 +1,3915 @@
+    loop = loops[0].children[3].children[0]
+    parent = loop.parent
+    loop.detach()
+    tdir.children[0].addchild(loop)
+    parent.addchild(tdir, index=0)
+    strans.apply(tree.children[0].children[:])
+    ptrans.apply(tree.children[0].children[:])
+    correct = '''subroutine my_subroutine()
+  type :: y
+    integer, dimension(321) :: jp
+  end type y
+  type :: x
+    type(y) :: y
+  end type x
+  integer :: i
+  integer :: ii
+  integer :: j
+  integer :: jj
+  integer :: k
+  type(x) :: ty
+
+  !$omp parallel default(shared), private(i,ii,j,jj)
+  !$omp single
+  do i = 1, 320, 32
+    !$omp task private(j,ii,jj), firstprivate(i), shared(k,ty), depend(in: \
+k,ty%y%jp(i),ty%y%jp(i + 32),ty%y%jp(1)), depend(out: k,ty%y%jp(i + 32),\
+ty%y%jp(i),ty%y%jp(1))
+    do j = 1, 320, 32
+      do ii = i, i + 32, 1
+        do jj = j, j + 32, 1
+          k = k + ty%y%jp(ii) + ii
+          ty%y%jp(ii + 1) = ty%y%jp(ii + 1) - (1 - ty%y%jp(ii + 1))
+          ty%y%jp(1) = ty%y%jp(1) + 1
+        enddo
+      enddo
+    enddo
+    !$omp end task
+  enddo
+  !$omp end single
+  !$omp end parallel
+
+end subroutine my_subroutine\n'''
+    assert fortran_writer(tree) == correct
+
+
+def test_omp_task_directive_32(fortran_reader):
+    ''' Test the code generation correctly generates the correct code if
+    code contains a type within a type .'''
+    code = '''
+    subroutine my_subroutine()
+      type :: y
+        integer, dimension(321) :: jp
+      end type
+      type :: x
+        type(y), dimension(5) :: y
+      end type
+        integer :: i, ii
+        integer :: j, jj
+        integer :: k
+        type(x) :: ty
+
+        do i = 1, 320, 32
+            do j = 1, 320, 32
+                do ii=i, i+32
+                    do jj = j,j+32
+                        k = ty%y%jp(ii) + ii
+                        ty%y(2)%jp(ii+1) = ty%y(1)%jp(ii+1) - (1 - \
+ty%y(3)%jp(ii+1))
+                    end do
+                end do
+            end do
+        end do
+    end subroutine
+    '''
+    tree = fortran_reader.psyir_from_source(code)
+    ptrans = OMPParallelTrans()
+    strans = OMPSingleTrans()
+    tdir = DynamicOMPTaskDirective()
+    loops = tree.walk(Loop)
+    loop = loops[0].children[3].children[0]
+    parent = loop.parent
+    loop.detach()
+    tdir.children[0].addchild(loop)
+    parent.addchild(tdir, index=0)
+    strans.apply(tree.children[0].children[:])
+    ptrans.apply(tree.children[0].children[:])
+    with pytest.raises(GenerationError) as excinfo:
+        tree.lower_to_language_level()
+    assert("PSyclone doesn't support an OMPTaskDirective containing a "
+           "StructureReference with multiple "
+           "array accessing members. Found 'ty%y(2)%jp(ii + 1)'"
+           in str(excinfo.value))
+
+
+def test_omp_task_directive_write_index_shared_type(fortran_reader):
+    ''' Test the code generation correctly generates an error if an array index
+    of a written array is a shared variable.'''
```

Comment:
Please could you put this text next to the one that does this for a read array.

Reply:
Moved together.